homeassistant/components/tplink/config_flow.py

```diff
@@ -129,6 +293,25 @@ async def async_step_pick_device(
             data_schema=vol.Schema({vol.Required(CONF_DEVICE): vol.In(devices_name)}),
         )
 
+    async def _async_reload_requires_auth_entries(self) -> None:
+        """Reload any in progress config flow that now have credentials."""
+        _config_entries = self.hass.config_entries
+        for flow in _config_entries.flow.async_progress_by_handler(
+            DOMAIN, include_uninitialized=True
+        ):
+            context: dict[str, Any] = flow["context"]
+            if context.get("source") != SOURCE_REAUTH:
+                continue
+            entry_id: str = context["entry_id"]
+            if (
+                reauth_entry := self.reauth_entry
+            ) and entry_id == reauth_entry.entry_id:
+                continue  # pragma: no cover
```

Comment:
We need to figure out how to cover this line or refactor if it's truly unreachable. HA standards require 100% coverage for config flows.

Reply:
So this it line I think is unreachable because the reauth flow that calls ` _async_reload_requires_auth_entries` has already reloaded itself and aborted the flow by the time the task executes. Let me know if you disagree.

Reply:
I agree its unreachable.

```diff
diff --git a/homeassistant/components/tplink/config_flow.py b/homeassistant/components/tplink/config_flow.py
index 45afbbb0346..52e973fc9f6 100644
--- a/homeassistant/components/tplink/config_flow.py
+++ b/homeassistant/components/tplink/config_flow.py
@@ -295,6 +295,9 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     async def _async_reload_requires_auth_entries(self) -> None:
         """Reload any in progress config flow that now have credentials."""
+        if (reauth_entry := self.reauth_entry):
+            await self.hass.config_entries.async_reload(reauth_entry.entry_id)
+
         _config_entries = self.hass.config_entries
         for flow in _config_entries.flow.async_progress_by_handler(
             DOMAIN, include_uninitialized=True
@@ -303,10 +306,6 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
             if context.get("source") != SOURCE_REAUTH:
                 continue
             entry_id: str = context["entry_id"]
-            if (
-                reauth_entry := self.reauth_entry
-            ) and entry_id == reauth_entry.entry_id:
-                continue  # pragma: no cover
             if entry := _config_entries.async_get_entry(entry_id):
                 await _config_entries.async_reload(entry.entry_id)
                 if entry.state is ConfigEntryState.LOADED:
@@ -415,7 +414,6 @@ class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
                 errors["base"] = "cannot_connect"
             else:
                 await set_credentials(self.hass, username, password)
-                await self.hass.config_entries.async_reload(reauth_entry.entry_id)
                 self.hass.async_create_task(self._async_reload_requires_auth_entries())
                 return self.async_abort(reason="reauth_successful")
 
```

Suggested change to reload the config entry in the task to ensure it doesn't deadlock